include/openenclave/corelibc/netdb.h

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+/* DISCLAIMER:
+ * This header is published with no guarantees of stability and is not part
+ * of the Open Enclave public API surface. It is only intended to be used
+ * internally by the OE runtime. */
+
+#ifndef _OE_NETDB_H
+#define _OE_NETDB_H
+
+#include <openenclave/internal/syscall/netdb.h>
+
+OE_EXTERNC_BEGIN
+
+/*
+**==============================================================================
+**
+** OE names:
+**
+**==============================================================================
+*/
+
+const char* oe_gai_strerror(int errnum);
+
+/*
+**==============================================================================
+**
+** Standard-C names:
+**
+**==============================================================================
+*/
+
+#if defined(OE_NEED_STDC_NAMES)
+
+OE_INLINE
+const char* gai_strerror(int errnum)
```

Comment:
Does this need to be part of corelibc? The other socket functionality it part of oelibc (see the `libc/` directory).

I would recommend adding the function at that level and just compiling in the source file from musl. You can follow the same model that is used for compiling socket.c in `libc/CMakeLists.txt`

Reply:
Is there a design doc somewhere that explains the difference between oelibc and corelibc?   (I can guess, but is it documented somewhere?)

Reply:
I don't think there is.

Basically there used to only be corelibc and the goal was to implement a minimal libc there. Then it it was decided that that was challenging, not worth the trouble, and using an existing libc implementation was better. So musl was added. But now oecore depends on some basic stdc functions (mostly string and memory related) so corelibc couldn't be entirely removed.

So for the most part new libc functionality all goes in oelibc. There is also probably a lot of things in corelibc that aren't needed anymore and need to be cleaned up.

Reply:
Filed issue #2907.  I did some more testing, and gai_strerror is already part of oelibc so I will abandon this PR.